app-creator/client/src/widgets/tool-bar/tool-bar.ts

```diff
@@ -152,38 +201,139 @@ export class ToolBar extends LitElement {
     textArea.remove();
   }
 
+  /**
+   * Imports template from JSON provided by the user.
+   */
+  importTemplate() {
+    // Get textarea input element.
+    if (this.importTextArea == null) {
+      return;
+    }
+
+    // Get dialog element.
+    if (this.importDialog == null) {
+      return;
+    }
+
+    // Get template json string.
+    const template = this.importTextArea.value.replace(/\\'/g, "'").trim();
```

Comment:
The user may accidentally include more than they intended, which we can resolve for the user using string manipulation to parse out the JSON part. Perhaps something as simple as (without thinking through the edge cases):
```js
template = template.slice(template.indexOf('{'), template.lastIndexOf('}'));
```